### ISSUE TITLE
Checking results status in Python

### ISSUE THREAD

**Issue by stephane-caron** [oxfordcontrol/Clarabel.rs#10]
Thanks for releasing Clarabel! As I was trying it out in Python, I noticed something confusing about the status of a result: currently it can be both ``==`` and ``!=`` to ``Solved`` at the same time.

Here is a small example, following: https://oxfordcontrol.github.io/ClarabelDocs/stable/python/getting_started_py/

```python
    solver = clarabel.DefaultSolver(P, q, A, b, cones, settings)
    result = solver.solve()
    print(type(result.status))
    print(result.status == clarabel.SolverStatus.Solved)
    print(result.status != clarabel.SolverStatus.Solved)
    print(result.status is clarabel.SolverStatus.Solved)  # to check the underlying references
```

The output on my machine is:

```console
<class 'builtins.SolverStatus'>
True
True
False
```

Clarabel was installed from PyPI. Version:

```python
>>> import clarabel
>>> clarabel.__version__
<built-in function __version__>  # a bit confusing, __version__ is usually a string in Python
>>> clarabel.__version__()
'0.3.0'
```

**Comment by goulart-paul** [comment]
I think that the issue is the same as the one described here:  https://github.com/PyO3/pyo3/issues/2618

A made a fix #11 by bumping the Py03 version, which seems to solve the problem for me locally.   We will merge this and distribute update binaries as part of v0.4, which should be coming soon.

**Comment by stephane-caron** [comment]
Okay :ok_hand: In the meantime for fellow Python travelers, a temporary workaround can be something like:

```python
    #if result.status != clarabel.SolverStatus.Solved:
    # Hack to go around https://github.com/oxfordcontrol/Clarabel.rs/issues/10
    if str(result.status) != "Solved":
```

Looking forward to v0.4 :eyes:
